vendor/bat-native-ledger/src/bat/ledger/internal/contribution/unverified.cc

```diff
@@ -23,11 +23,24 @@ Unverified::Unverified(LedgerImpl* ledger) :
 Unverified::~Unverified() = default;
 
 void Unverified::Contribute() {
-  ledger_->wallet()->FetchBalance(
-      std::bind(&Unverified::OnContributeUnverifiedBalance,
-                this,
-                _1,
-                _2));
+  ledger_->database()->GetUnverifiedPublishersForPendingContributions(
+      std::bind(&Unverified::FetchInfoForUnverifiedPublishers, this, _1));
+}
+
+void Unverified::FetchInfoForUnverifiedPublishers(
+    std::vector<std::string>&& publisher_keys) {
+  if (!publisher_keys.empty()) {
+    const auto publisher_key = std::move(publisher_keys.back());
+    ledger_->publisher()->FetchServerPublisherInfo(
+        publisher_key, [this, publisher_keys = std::move(publisher_keys)](
+                           type::ServerPublisherInfoPtr) mutable {
+          publisher_keys.pop_back();
+          FetchInfoForUnverifiedPublishers(std::move(publisher_keys));
+        });
+  } else {
+    ledger_->wallet()->FetchBalance(
+        std::bind(&Unverified::OnContributeUnverifiedBalance, this, _1, _2));
```

Comment:
We shouldn't introduce `std::bind` into any new code. I know this is just following the established pattern in these files, but it came up a few times during the adaptive captcha review process. AFAIK, we can freely use `base::Bind*` in Rewards now.

Reply:
Is that possible without rewriting `wallet()->FetchBalance(...)`?

Reply:
Right, probably not. If that's a lot more effort than makes sense for this PR, I'm personally fine with skipping it in this particular case. Just wanted to make sure to flag it as a potential problem area.